contracts/nft_royalty/lib.rs

```diff
@@ -0,0 +1,437 @@
+            );
+            debug_println!("Creating an Instruction. Legs: {:?}", legs);
+            self.api
+                .add_and_affirm_instruction(venue_id, legs.clone(), vec![royalty_portfolio])?;
+            Ok(legs)
+        }
+
+        /// Creates a portoflio owned by the contract's caller and transfer its custody to the smart contract.
+        #[ink(message)]
+        pub fn create_custody_portfolio(&mut self, portfolio_name: PortfolioName) -> Result<()> {
+            self.contract_identity = PolymeshInk::get_our_did()?;
+            let callers_identity = Self::get_callers_identity()?;
+
+            if self.royalty_portfolios.contains(callers_identity) {
+                return Err(Error::RoyaltyPortfolioAlreadyExists);
+            }
+
+            let portfolio_id = self
+                .api
+                .create_custody_portfolio(callers_identity, portfolio_name)?;
+
+            self.royalty_portfolios
+                .insert(callers_identity, &portfolio_id);
+            Ok(())
+        }
+
+        /// Ensures the metadata rules are being respected in the transfer
+        fn ensure_valid_transfer_values(
+            nft_artist_rules: &NFTArtistRules,
+            nft_offer: &NFTOffer,
+        ) -> Result<()> {
+            if !nft_artist_rules.is_ticker_allowed(&nft_offer.purchase_ticker) {
+                return Err(Error::TickerNotAllowedForRoyalty(nft_offer.purchase_ticker));
+            }
+            Ok(())
+        }
+
+        /// Returns a [`Vec<Leg>`] for an instruction transfering an NFT.
+        fn setup_legs(
+            &mut self,
+            nft_transfer_details: NFTTransferDetails,
+            nft_offer: NFTOffer,
+            royalty_percentage: Perbill,
+            royalty_portfolio: PortfolioId,
+        ) -> Vec<Leg> {
+            // The first leg transfers the NFT to the buyer
+            let nfts = NFTs {
+                ticker: nft_transfer_details.collection_ticker,
+                ids: nft_transfer_details.nfts,
+            };
+            let nft_leg = Leg::NonFungible {
+                sender: nft_transfer_details.nft_owner_portfolio,
+                receiver: nft_transfer_details.nft_receiver_portfolio,
+                nfts,
+            };
+            // The second leg transfers the payment to the seller
+            let nft_payment_leg = Leg::Fungible {
+                sender: nft_offer.payer_portfolio,
+                receiver: nft_offer.receiver_portfolio,
+                ticker: nft_offer.purchase_ticker,
+                amount: nft_offer.transfer_price,
+            };
+            // The third leg transfers the royalty to the artist
+            let royalty_amount = royalty_percentage * nft_offer.transfer_price;
+            let royalty_leg = Leg::Fungible {
+                sender: nft_offer.payer_portfolio,
+                receiver: royalty_portfolio,
+                ticker: nft_offer.purchase_ticker,
+                amount: royalty_amount,
```

Comment:
Maybe the `royalty_amount` should be deducted from the `amount` sent to the seller.

Reply:
For example:

1. Seller sets the price at 100 USDC.
2. With a royalty of 20%.
3. 80 USDC would go to the seller and 20 USDC would go to the artist.